### DISCUSSION TITLE
[Solved] NoConfigFoundError

### DISCUSSION THREAD

**Discussion by nelsonlaidev** [vercel/community#443] [Help] [answered]
Although I don't know the reason, the problem was resolved.

---

Hi, there!

When I deploy my next.js project to vercel, I got this error. What is that meaning? I feel confused. Sorry, I don't know what made this error. Is it next.js or vercel? I can't find any information about this error on the internet.

```
18:22:42.160 | > tszhong-blog@1.0.6 build /vercel/path0
18:22:42.160 | > next build
18:22:42.160 |  
18:22:45.380 | NoConfigFoundError {
18:22:45.380 | cwd: '/vercel/path0',
18:22:45.380 | configPath: undefined,
18:22:45.380 | _tag: 'NoConfigFoundError',
18:22:45.381 | toString: [Function (anonymous)],
18:22:45.381 | [Symbol()]: { cwd: '/vercel/path0', configPath: undefined },
18:22:45.381 | [Symbol()]: [ 'configPath', 'cwd' ]
18:22:45.381 | }
18:22:45.395 | npm ERR! code ELIFECYCLE
18:22:45.395 | npm ERR! errno 1
18:22:45.398 | npm ERR! tszhong-blog@1.0.6 build: `next build`
18:22:45.398 | npm ERR! Exit status 1
18:22:45.398 | npm ERR!
18:22:45.398 | npm ERR! Failed at the tszhong-blog@1.0.6 build script.
18:22:45.398 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
18:22:45.412 |  
18:22:45.413 | npm ERR! A complete log of this run can be found in:
18:22:45.413 | npm ERR!     /vercel/.npm/_logs/2022-02-25T10_22_45_399Z-debug.log
18:22:45.419 | Error: Command "npm run build" exited with 1
```

**Comment by samcx** [comment, marked as answer]
Marking this as solved!

**Reply by peduarte** [reply]
im having the same issue 😅 what was the solution?

**Reply by Itotutona** [reply]
I'm also having the same issue. Could you explain the solution ?

**Reply by peduarte** [reply]
https://github.com/vercel/community/discussions/443#discussioncomment-2360438

**Reply by AntonyJudeShaman** [reply]
hey whats the solution for this. please help me

**Comment by peduarte** [comment]
after a long time debugging and with @schickling help we discovered it was related to `pnpm`.

the fix in my case was to change the build scrips as below:
```diff
- "build": "next build"
+ "build": "INIT_CWD=$PWD next build"
```

related: https://github.com/contentlayerdev/contentlayer/issues/104#issuecomment-1026697771

**Reply by Itotutona** [reply]
Thanks, i will try it quickly

**Reply by alamenai** [reply]
unfortunately, it did not work for me.

**Reply by verdverm** [reply]
It's almost certainly a config issue with ContentLayer, at least for me.

Does the `contentlayer.config.ts` exist for you?
